cachi2/core/package_managers/pip.py

```diff
@@ -87,11 +89,16 @@ def fetch_pip_source(request: Request) -> RequestOutput:
             package.requirements_files,
             package.requirements_build_files,
         )
-
-        components.append(Component.from_package_dict(info["package"]))
+        purl = _generate_purl(info["package"], request.source_dir.join_within_root(package.path))
+        components.append(
+            Component(name=info["package"]["name"], version=info["package"]["version"], purl=purl)
+        )
 
         for dependency in info["dependencies"]:
-            components.append(Component.from_package_dict(dependency))
+            purl = _generate_purl(dependency)
+            components.append(
+                Component(name=dependency["name"], version=dependency["version"], purl=purl)
```

Comment:
Could we make sure to pass only semver-like versions here (the versions of "pypi" dependencies)? The existing hack was meant to be temporary https://github.com/containerbuildsystem/cachi2/blob/9917e6f3169bf74e661da1501417b8048a162204/cachi2/core/models/output.py#L37-L46

Reply:
added check for semver-like version and removed `_valid_semver` method